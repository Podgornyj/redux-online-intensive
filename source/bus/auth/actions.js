import { types } from "./types";

export const authActions = {
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },
    authenticateAsync: () => {
        return {
            type: types.AUTHENTICATE_ASYNC,
        };
    },
    signupAsync: (userData) => {
        return {
            type: types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (userData) => {
        return {
            type: types.LOGIN_ASYNC,
            payload: userData,
        };
    },
    initialize: () => {
        return {
            type: types.INITIALIZE,
        };
    },
    initializeAsync: () => {
        return {
            type: types.INITIALIZE_ASYNC,
        };
    },
    logout: () => {
        return {
            type: types.LOGOUT,
        };
    },
    logoutAsync: () => {
        return {
            type: types.LOGOUT_ASYNC,
        };
    },
};
