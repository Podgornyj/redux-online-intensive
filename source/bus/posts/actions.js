import { types } from './types';
import { api } from '../../REST';


export const  postsActions = {
    fillPosts : (posts) => {
        return {
            type: types.FILL_POSTS,
            payload: posts,
        };
    },
    fetchPostsAsync : () => {
        return {
            type: types.FETCH_POST_ASYNC,
        };
    },
    fillNewPost : (post) => {
        return {
            type: types.FILL_NEW_POST,
            payload: post,
        };
    },
    createPostAsync : (comment) => {
        return {
            type: types.CREATE_POST_ASYNC,
            payload: comment,
        };
    },
};
