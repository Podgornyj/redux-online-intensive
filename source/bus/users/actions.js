import { types } from "./types";

export const userActions = {
    clearUsers: () => {
        return {
            type: types.CLEAR_USERS,
        };
    },
    fillUsers: (users) => {
        return {
            type: types.FILL_USERS,
            payload: users,
        };
    },
    fetchUsersAsync: () => {
        return {
            type: types.FETCH_USERS_ASYNC,
        };
    },
};
