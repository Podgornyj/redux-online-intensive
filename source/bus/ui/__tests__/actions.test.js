import { uiActions } from '../actions';

test('Test startFetching action', () => {
    expect(uiActions.startFetching()).toMatchSnapshot();
});
test('Test stoptFetching action', () => {
    expect(uiActions.stoptFetching()).toMatchSnapshot();
});
test('Test setOnlineState action', () => {
    expect(uiActions.setOnlineState()).toMatchSnapshot();
});
test('Test setOfflineState action', () => {
    expect(uiActions.setOfflineState()).toMatchSnapshot();
});
test('Test emitError action', () => {
    expect(uiActions.emitError(__.error, __.errorMessage)).toMatchSnapshot();
});
