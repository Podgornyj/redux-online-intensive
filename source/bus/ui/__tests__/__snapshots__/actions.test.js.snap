// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test emitError action 1`] = `
Object {
  "error": true,
  "meta": "TEST_ERROR_MESSAGE.",
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`Test setOfflineState action 1`] = `
Object {
  "type": "SET_OFFLINE_STATE",
}
`;

exports[`Test setOnlineState action 1`] = `
Object {
  "type": "SET_ONLINE_STATE",
}
`;

exports[`Test startFetching action 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`Test stoptFetching action 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
